lib/OpenQA/WebAPI/Controller/API/V1/Search.pm

```diff
@@ -147,12 +147,22 @@ sub _search_job_templates {
     my ($self, $keywords, $limit) = @_;
 
     my @results;
-    my $last_group = -1;
-    my $like       = {like => "%${keywords}%"};
-    my $templates  = $self->schema->resultset('JobTemplates')->search({-or => {name => $like, description => $like}})
-      ->slice(0, $limit);
+    my $last_group     = -1;
+    my $like           = {like => "%${keywords}%"};
+    my $test_suite_ids = $self->_get_test_suite_ids($keywords);
+    my $templates
+      = $self->schema->resultset('JobTemplates')
+      ->search({-or => {name => $like, description => $like, test_suite_id => {-in => $test_suite_ids}}},
+        {order_by => {-desc => 'group_id'}})->slice(0, $limit);
```

Comment:
Wouldn't it be nice to join with the test suites table here instead of using 2 different queries?

Reply:
Hm, I do not know how to do that, could you give me an example? Thanks.

Reply:
I phrased it as a question because maybe there are some pitfalls ahead. However, theoretically the join could be written like the one from this function: https://github.com/os-autoinst/openQA/blob/master/lib/OpenQA/Schema/ResultSet/Jobs.pm#L519

Reply:
I tried this 
```
= $self->schema->resultset('JobTemplates')
      ->search({-or => [name => $like, description => $like, 'test_suite.name' => $like]},
        {
            join    => 'test_suite',
            order_by => {-desc => 'group_id'}})->slice(0, $limit);
```
But it fails with the error:
```
[2020-09-16T20:09:55.0488 CST] [error] [ogLH2VQp] DBIx::Class::Storage::DBI::_dbh_execute(): DBI Exception: DBD::Pg::st execute failed: ERROR:  column reference "name" is ambiguous
LINE 1: ...ite ON test_suite.id = me.test_suite_id WHERE ( ( name LIKE ...
                                                             ^ [for Statement "SELECT me.id, me.product_id, me.machine_id, me.test_suite_id, me.name, me.description, me.prio, me.group_id, me.t_created, me.t_updated FROM job_templates me  JOIN test_suites test_suite ON test_suite.id = me.test_suite_id WHERE ( ( name LIKE ? OR description LIKE ? OR test_suite.name LIKE ? ) ) ORDER BY group_id DESC LIMIT ?" with ParamValues: 1='%apparmor%', 2='%apparmor%', 3='%apparmor%', 4='61'] at /home/liuxiaojing/code/repos/openQA/script/../lib/OpenQA/WebAPI/Controller/API/V1/Search.pm line 162
```
Do you have any idea? I do not know how to deal with the `or` relation using `join`. Thanks.

Reply:
Likely you just need to use `'me.name'` instead of `name` in the `{-or => …}` hash. The same might count for the description as well.